source/common/config/grpc_mux_subscription_impl.cc

```diff
@@ -93,6 +94,19 @@ void GrpcMuxSubscriptionImpl::onConfigUpdateFailed(ConfigUpdateFailureReason rea
   callbacks_.onConfigUpdateFailed(reason, e);
 }
 
+void GrpcMuxSubscriptionImpl::onTryFallback(ConfigUpdateFailureReason reason) {
+  assert(reason == Envoy::Config::ConfigUpdateFailureReason::ConnectionFailure);
+  stats_.update_failure_.inc();
+  ENVOY_LOG(debug, "gRPC update for {} failed", type_url_);
+  stats_.update_attempt_.inc();
+  callbacks_.kickFallback();
+}
+
+void GrpcMuxSubscriptionImpl::fallback(const std::set<std::string>& resources) {
+  type_url_ = TypeUrl::get().fallback(type_url_);
```

Comment:
If we're doing resource-level fallback, shouldn't we be able to infer this form the API type oracle?